src/main/java/com/uber/profiling/Arguments.java

```diff
@@ -152,71 +156,114 @@ public void updateArguments(Map<String, List<String>> parsedArgs) {
             throw new RuntimeException("Sample interval too short, must be 0 (disable sampling) or at least " + Arguments.MIN_INTERVAL_MILLIS);
         }
 
-        tag = getArgumentSingleValue(parsedArgs, ARG_TAG);
-        logger.info("Got argument value for tag: " + tag);
+        argValue = getArgumentSingleValue(parsedArgs, ARG_TAG);
+        if (needToUpdateArg(argValue)) {
+            tag = argValue;
+            logger.info("Got argument value for tag: " + tag);
+        }
 
         argValue = getArgumentSingleValue(parsedArgs, ARG_APP_ID_REGEX);
-        if (argValue != null && !argValue.isEmpty()) {
+        if (needToUpdateArg(argValue)) {
             appIdRegex = argValue;
             logger.info("Got argument value for appIdRegex: " + appIdRegex);
         }
 
         List<String> argValues = getArgumentMultiValues(parsedArgs, ARG_DURATION_PROFILING);
-        for (String str : argValues) {
-            int index = str.lastIndexOf(".");
-            if (index <= 0 || index + 1 >= str.length()) {
-                throw new IllegalArgumentException("Invalid argument value: " + str);
+        if (!argValues.isEmpty()) {
+            durationProfiling.clear();
+            for (String str : argValues) {
+                int index = str.lastIndexOf(".");
+                if (index <= 0 || index + 1 >= str.length()) {
+                    throw new IllegalArgumentException("Invalid argument value: " + str);
+                }
+                String className = str.substring(0, index);
+                String methodName = str.substring(index + 1, str.length());
+                ClassAndMethod classAndMethod = new ClassAndMethod(className, methodName);
+                durationProfiling.add(classAndMethod);
+                logger.info("Got argument value for durationProfiling: " + classAndMethod);
             }
-            String className = str.substring(0, index);
-            String methodName = str.substring(index + 1, str.length());
-            ClassAndMethod classAndMethod = new ClassAndMethod(className, methodName);
-            durationProfiling.add(classAndMethod);
-            logger.info("Got argument value for durationProfiling: " + classAndMethod);
         }
 
         argValues = getArgumentMultiValues(parsedArgs, ARG_ARGUMENT_PROFILING);
-        for (String str : argValues) {
-            int index = str.lastIndexOf(".");
-            if (index <= 0 || index + 1 >= str.length()) {
-                throw new IllegalArgumentException("Invalid argument value: " + str);
-            }
-            String classMethodName = str.substring(0, index);
-            int argumentIndex = Integer.parseInt(str.substring(index + 1, str.length()));
+        if (!argValues.isEmpty()) {
```

Comment:
use needToUpdateArg? argValues cannot be null any way

Reply:
argValues is a list here so just check empty. needToUpdateArg is to check a string value.